vspace/src/main/java/edu/asu/diging/vspace/core/data/IdGenerator.java

```diff
@@ -12,20 +12,27 @@
 import org.hibernate.type.Type;
 
 public class IdGenerator implements IdentifierGenerator, Configurable {
-	
-	private String prefix;
 
-	@Override
-	public void configure(Type type, Properties properties, ServiceRegistry sr) throws MappingException {
-		prefix = properties.getProperty("prefix");
-	}
+    private String prefix;
 
-	@Override
-	public Serializable generate(SharedSessionContractImplementor session, Object obj) throws HibernateException {
-		String query = String.format("select count(*) from %s",
-	            obj.getClass().getSimpleName());
-		long count = (Long) session.createQuery(query).uniqueResult();
-		return prefix + (count + 1);
-	}
+    @Override
+    public void configure(Type type, Properties properties, ServiceRegistry sr) throws MappingException {
+        prefix = properties.getProperty("prefix");
+    }
 
+    @Override
+    public Serializable generate(SharedSessionContractImplementor session, Object obj) throws HibernateException {
+        int value = 0;
+        String query = String.format("SELECT id FROM %s ORDER BY creationDate DESC", obj.getClass().getSimpleName());
+        String prevId = (String) session.createQuery(query).setMaxResults(1).uniqueResult();
+        if (prevId == null) {
+            value = 1;
+            String prefix = (String) this.prefix.subSequence(0, this.prefix.length() - 9);
+            return prefix + String.format("%09d", value);
+        } else {
+            value = Integer.parseInt(prevId.substring(prevId.length() - 9)) + 1;
+            String prefix = (String) prevId.subSequence(0, prevId.length() - 9);
+            return prefix + String.format("%09d", value);
+        }
+    }
```

Comment:
Sorry, I realized that this can give us issues, especially when scaling horizontally. Can you change it to how it is implemented here (keeping "%09d" is fine though):
https://github.com/diging/citesphere/blob/develop/citesphere/src/main/java/edu/asu/diging/citesphere/core/repository/IdGenerator.java